packages/web/src/generic.test.ts

```diff
@@ -39,3 +39,82 @@ describe('inject', () => {
     });
   });
 });
+
+describe('track custom events', () => {
+  beforeEach(() => {
+    // reset the internal queue before every test
+    window.vaq = [];
+  });
+  describe('queue custom events', () => {
+    it('event name only', () => {
+      inject();
+
+      track('my event');
+
+      expect(window.vaq).toBeDefined();
+
+      if (!window.vaq) throw new Error('window.vaq is not defined');
+
+      expect(window.vaq[0]).toEqual([
+        'track',
+        {
+          name: 'my event',
+        },
+      ]);
+    });
+    it('with custom data', () => {
+      inject();
+
+      track('custom event', {
+        string: 'string',
+        number: 1,
+      });
+
+      expect(window.vaq).toBeDefined();
+
+      if (!window.vaq) throw new Error('window.vaq is not defined');
+
+      expect(window.vaq[0]).toEqual([
+        'track',
+        {
+          name: 'custom event',
+          data: {
+            string: 'string',
+            number: 1,
+          },
+        },
+      ]);
+    });
+
+    it('strip nested object', () => {
+      process.env = {
+        NODE_ENV: 'production',
+      };
+      inject();
+
+      track('custom event', {
+        string: 'string',
+        number: 1,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        nested: {
+          object: '',
+        } as any,
+      });
+
+      expect(window.vaq).toBeDefined();
+
+      if (!window.vaq) throw new Error('window.vaq is not defined');
```

Comment:
I think this is redundant as we already have an expect for that on line 104

Reply:
@chriswdmr it satisfies typescript